### ISSUE TITLE
SilentPop is broken

### ISSUE THREAD

**Issue by bobbybonifacia** [exrin/Exrin#5]
When we call _navigationService.SilentPop, in the NavigationProxy.SilentPopAsync() we receive a wrong index.
I think, the wrong collection is picked for current stack views indexing: _viewKeyTracking (represents the current stack, as far as I understand) should be used instead of _viewsByKey(represents the views mapped in the current stack) in BaseStack.SilentPop(). Also, after proxy.SilentPopAsync() is called, the "popped" view should be removed from _viewKeyTracking list.
I changed method code in BaseStack.cs to this:
```
public async Task SilentPop(IList<string> viewKeys)
{
  foreach (var key in viewKeys)
  {
    var tuple = Abstraction.Tuple.Create(key, App.PlatformOptions?.Platform);
    // Get index in stack
    var index = _viewKeyTracking.IndexOf(_viewKeyTracking.FirstOrDefault(x => x.Key == key));
    var indexFromTop = _viewKeyTracking.Count - index - 1;
    await Proxy.SilentPopAsync(indexFromTop);
    //change the stack
    _viewKeyTracking.RemoveAt(index);
  }
}
```
And now it works as expected.

**Comment by adamped** [comment]
@bobbybonifacia - thanks for reporting this. I will look into it and test it shortly, and push out a fix.

**Comment by adamped** [comment]
@bobbybonifacia - this has been fixed in 1.0.5. Please re-open if you continue to experience issues.
